execution/binary/scalar.go

```diff
@@ -103,11 +94,21 @@ func (o *scalarOperator) Next(ctx context.Context) ([]model.StepVector, error) {
 		return nil, err
 	}
 
+	scalarIn, err := o.scalar.Next(ctx)
+	if err != nil {
+		return nil, err
+	}
+
 	out := o.pool.GetVectorBatch()
-	for _, vector := range in {
+	for v, vector := range in {
 		step := o.pool.GetStepVector(vector.T)
 		for i := range vector.Samples {
-			operands := o.getOperands(vector, i, o.scalar)
+			scalarVal := math.NaN()
```

Comment:
Do we need to handle the case where `scalarVal` remains NaN (when the if condition is false)?

Reply:
Yes, this is already being handled. In case, the if condition is false the result should be NaN for each step vector. I'll add test cases once https://github.com/efficientgo/core/pull/3 is merged!